### PR TITLE
Feat/4263  publication namespace

### PR DESCRIPTION
**Pull request by asf-algoo**
See #4263.

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/tracim_backend/migration/env.py [hunk 1]
```diff
@@ -41,6 +45,71 @@
 # ... etc.
 
 
+# Custom operations definitions
```
Comment (raphj) [inline backend/tracim_backend/migration/env.py:48]:
```suggestion
# Custom operations
```

or

```suggestion
# Definitions of custom operations
```

File: backend/tracim_backend/migration/versions/78a01733957f_add_publication_namespace.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+revision = "78a01733957f"
+down_revision = "5d54d8602f5a"
+
+old_enum_values = ("CONTENT", "UPLOAD")
+old_enum = sa.Enum(*old_enum_values, name="contentnamespaces")
+new_enum = sa.Enum(*(old_enum_values + ("PUBLICATION",)), name="contentnamespaces")
+
+
+def upgrade() -> None:
+    op.replace_enum("content_revisions", "content_namespace", old_enum, new_enum, None)
```
Comment (inkhey) [inline backend/tracim_backend/migration/versions/78a01733957f_add_publication_namespace.py:21]: very nice solution :+1:

**Review by raphj — APPROVED** [review]

File: backend/tracim_backend/tests/functional/test_workspaces.py [hunk 3]
```diff
@@ -4112,7 +4115,13 @@ def test_api__post_content_create_generic_content__ok_200__nominal_case(
         assert res.json_body["created"]
         assert res.json_body["file_extension"] == ".document.html"
         assert res.json_body["filename"] == "GenericCreatedContent.document.html"
-        params_active = {"parent_ids": 0, "show_archived": 0, "show_deleted": 0, "show_active": 1}
+        params_active = {
```
Comment (inkhey) [inline backend/tracim_backend/tests/functional/test_workspaces.py:4118]: Does this mean previous http request done by the frontend will failed or namespace_filter is optional ?

**Review by inkhey — APPROVED** [review]

File: backend/tracim_backend/tests/functional/test_workspaces.py (hunk 3 above)
Comment (grignards) [reply backend/tracim_backend/tests/functional/test_workspaces.py:4118]: namespaces_filter is optional (I didn't added it, it already existed probably for listing guest uploads).

**Pull request merged**